### PR TITLE
Fix/orders receive key address

### PR DESCRIPTION
**Pull request by mn13**
Previous address receiving ignores compressed public keys. In this PR address received with derivedAddress param in `keyById`.
To convert string to address in solidity used [oppenzeppelin's Strings.sol](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/release-v5.2/contracts/utils/Strings.sol#L334) lib. As it is not released yet I just moved used functions into our project

## Summary by CodeRabbit

- **New Features**
	- Introduced a library for string operations, enabling parsing of hexadecimal strings into Ethereum addresses.
	- Enhanced address derivation process in the `BasicOrder` contract.

- **Bug Fixes**
	- Updated transaction details and metadata for improved deployment accuracy.

- **Refactor**
	- Modified the `keyById` function to return a constructed `KeyResponse` instead of directly accessing a mapping.

- **Chores**
	- Updated the `.gitignore` to ignore all contents within a specific subdirectory.

### CODE REVIEW

**Review by backsapc — DISMISSED** [review]

File: solidity/orders/src/BasicOrder.sol [hunk 1]
```diff
@@ -83,8 +86,10 @@ contract BasicOrder is IExecution, ReentrancyGuard {
         }
 
         WARDEN_PRECOMPILE = IWarden(IWARDEN_PRECOMPILE_ADDRESS);
-        KeyResponse memory keyResponse = WARDEN_PRECOMPILE.keyById(_orderData.signRequestData.keyId, new int32[](0));
-        _keyAddress = address(bytes20(keccak256(keyResponse.key.publicKey)));
+        int32[] memory deriveAddresses = new int32[](1);
+        deriveAddresses[0] = 1;
```
Comment (backsapc) [inline solidity/orders/src/BasicOrder.sol]: nit: Rename this to addressTypes

File: solidity/orders/src/BasicOrder.sol [hunk 2]
```diff
@@ -83,8 +86,10 @@ contract BasicOrder is IExecution, ReentrancyGuard {
         }
 
         WARDEN_PRECOMPILE = IWarden(IWARDEN_PRECOMPILE_ADDRESS);
-        KeyResponse memory keyResponse = WARDEN_PRECOMPILE.keyById(_orderData.signRequestData.keyId, new int32[](0));
-        _keyAddress = address(bytes20(keccak256(keyResponse.key.publicKey)));
+        int32[] memory deriveAddresses = new int32[](1);
```
Comment (backsapc) [inline solidity/orders/src/BasicOrder.sol]: nit: Extract `1` to a constant EthereumAddressType

File: solidity/orders/src/Strings.sol [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+// OpenZeppelin Contracts (last updated v5.1.0) (utils/Strings.sol)
+
+pragma solidity >=0.8.25 <0.9.0;
+
+import { SafeCast } from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+/**
+ * @dev String operations.
+ */
+library Strings {
```
Comment (Svetomech) [inline solidity/orders/src/Strings.sol:11]: question: why is this not released yet as part of the standard?

**Review by backsapc — APPROVED** [review]

**Review by Svetomech — APPROVED** [review]

File: solidity/orders/src/Strings.sol (hunk 3 above)
Comment (mn13) [reply solidity/orders/src/Strings.sol:11]: Do you mean not a release yet? I think guys from @OpenZeppelin waiting for audit before release. This methods appeared recently and I think will be released in next version

**Pull request merged**